### ISSUE TITLE
Play better with BugSee

### ISSUE THREAD

**Issue by nbrooke** [steamclock/steamclog-swift#83]
We use BugSee in some projects for test builds to more-invasive-but-more-useful error reporting, but Steamclog/Sentry doesn't always play well with it (can have problems if both Sentry and BugSee are on at the same time, if the project isn't using BugSee we probably DO want to be capturing Sentry info in test builds by default, etc.)

There's a broad range of potential solution here, ranging from just coming up with some guidelines here, all the way to potentially just integrating BugSee into Steamclog as well and making the default behaviour correct out-of-the-box.

**Comment by JeremyChiang** [comment]
Looks like we've already filed this ticket.

Here's a recent discussion on a crash discovery and later another suggestion to include to Bugsee:
https://steamclock.slack.com/archives/C0WL7G3JL/p1649969406222429

**Comment by JeremyChiang** [comment]
Hi @brendanlensink @jenncoop when you have time, I would like to get your feedback on this. Thanks! I could try to implement this for us.

**Comment by brendanlensink** [comment]
I'm a little hesitant to get behind integrating Bugsee straight into SteamcLog, given that we [just spent some time decoupling Sentry from SteamcLog](https://github.com/steamclock/steamclog-swift/pull/104). Maybe it's different enough as an internal tool that it's worth doing?

Maybe it would be worth looking at how we're setting up SteamcLog and making it more modular, then allowing users to set up integrations with things like Sentry/Bugsee using custom Destinations?

**Comment by nbrooke** [comment]
Yeah, I think we are actually a fair bit more likely to want an alternate production logging library (i.e. Sentry vs. Rollbar) than a more internal debugging tool like this, so I don't think that it's as important to make Bugsee replaceable, HOWEVER it's likely that any logging library that does crash handling is going to have that same issue that Sentry does of needing some co-ordination with BugSee at init time, so at least the way we handle the initialization needs to be modular enough that any replacement destination that might support crash reporting has some way to know that BugSee is active and turn off it's own crash reporting.

**Comment by sakuraehikaru** [comment]
I no longer think it’s a good idea to integrate Bugsee as well. I don’t know how yet, but I am also in favour of figuring out a way to make the ordering of init more resilient to crashing when Bugsee is involved.
